OutOfSchool/OutOfSchool.WebApi/Services/ProviderServices/ProviderService.cs

```diff
@@ -753,13 +753,16 @@ await UpdateWorkshopsProviderStatus(providerUpdateDto.Id, providerUpdateDto.Stat
             await actionBeforeDeleting(entity).ConfigureAwait(false);
         }
 
-        await providerRepository.Delete(entity).ConfigureAwait(false);
+        await providerRepository.RunInTransaction(async () =>
+        {
+            await providerRepository.Delete(entity).ConfigureAwait(false);
 
-        logger.LogInformation("Provider with Id = {Id} successfully deleted", id);
+            logger.LogInformation("Provider with Id = {Id} successfully deleted", id);
 
-        await userService.Delete(entity.UserId);
+            await userService.Delete(entity.UserId);
 
-        logger.LogInformation("User with Id = {entity.UserId} successfully deleted", entity.UserId);
+            logger.LogInformation("User with Id = {entity.UserId} successfully deleted", entity.UserId);
+        });
```

Comment:
(Logic, logs) There is potential of having inconsistent logs. Transaction could be rollbacked after execution of this lambda. As a result you could have logs stating that "X successfully deleted", but nothing changed in db. 
Please consider logging after the `await ...RunInTransaction(...)`. 

(Concurrency) `.ConfigureAwait()` is inconsistently used, either use it for both awaitables or remove it for both.

Reply:
Done